### ISSUE TITLE
tensornetwork/tn_keras/test_layer.py fails

### ISSUE THREAD

**Issue by mganahl** [google/TensorNetwork#618]
I noticed that on my machine the test
test_train[512-DenseExpander]
fails with 
```
dummy_data = (array([[6, 3, 7, ..., 8, 2, 8],
       [6, 3, 2, ..., 9, 3, 5],
       [6, 8, 0, ..., 6, 6, 3],
       ...,
       [6...
       [0.],
       [0.],
       [0.],
       [0.],
       [0.],
       [0.],
       [0.],
       [0.],
       [0.]]))
make_model = <tensorflow.python.keras.engine.sequential.Sequential object at 0x174dea828>

    def test_train(dummy_data, make_model):
      # Disable the redefined-outer-name violation in this function
      # pylint: disable=redefined-outer-name
      data, labels = dummy_data
      model = make_model
    
      model.compile(optimizer='adam',
                    loss='binary_crossentropy',
                    metrics=['accuracy'])
    
      # Train the model for 10 epochs
      history = model.fit(data, labels, epochs=10, batch_size=32)
    
      # Check that loss decreases and accuracy increases
      assert history.history['loss'][0] > history.history['loss'][-1]
>     assert history.history['accuracy'][0] < history.history['accuracy'][-1]
E     assert 0.5070000290870667 < 0.5019999742507935
```
this could be a precision issue (it consitently fails though on my machine). @bpenchas did you write the test? If so it would be great if you could have a quick look? thanks!

**Comment by Thenerdstation** [comment]
Looks like it might be the case of a bad seed.

I think the tests in the keras layers aren't deterministic. I also hit issues where two tests accidentally write to the same file. We should really take the time and upgrade them.

**Comment by Thenerdstation** [comment]
I think this should be fixed now. Can you test it on your end?
